app/models/hyrax/permission_template.rb

```diff
@@ -79,11 +79,16 @@ def source
 
     ##
     # A bit of an analogue for a `belongs_to :source_model` as it crosses from Fedora to the DB
-    # @return [AdminSet, ::Collection]
-    # @raise [Hyrax::ObjectNotFoundError] when neither an AdminSet or Collection is found
+    # @return [AdminSet, ::Collection, Hyrax::AdministrativeSet, Hyrax::PcmdCollection]
+    # @raise [Hyrax::ObjectNotFoundError] when neither an admin set or collection is found
     def source_model
-      ActiveFedora::Base.find(source_id)
-    rescue ActiveFedora::ObjectNotFoundError
+      if Hyrax.config.disable_wings
+        source
+      else
+        Hyrax.query_service.find_by_alternate_identifier(alternate_identifier: source_id,
+                                                         use_valkyrie: Hyrax.config.use_valkyrie?)
+      end
```

Comment:
When `use_valkyrie?` is `false`, `find_by_alternate_identifier` will return an `AdminSet` or `Collection` (ActiveFedora).  Otherwise, this returns a `Hyrax::AdministrativeSet` or `Hyrax::PcdmCollection` (Valkyrie).